client/controllers/events/curatedEvent.coffee

```diff
@@ -1,15 +1,48 @@
 Incidents = require '/imports/collections/incidentReports.coffee'
 UserEvents = require '/imports/collections/userEvents.coffee'
+Articles = require '/imports/collections/articles.coffee'
 #Allow multiple modals or the suggested locations list won't show after the loading modal is hidden
 Modal.allowMultiple = true
 
 Template.curatedEvent.onCreated ->
-  @editState = new ReactiveVar false
+  @editState = new ReactiveVar(false)
+  @loaded = new ReactiveVar(false)
+  @incidentIds = new Meteor.Collection(null)
+  userEventId = @data.userEventId
+  @subscribe "userEvent", @data.userEventId
+
+  @autorun =>
+    userEvent = UserEvents.findOne(userEventId)
+    if userEvent
+      document.title = "Eidr-Connect: #{userEvent.eventName}"
+      userEvent.incidents.forEach (incident) =>
+        @incidentIds.upsert id: incident.id,
+          id: incident.id
+
+  @autorun =>
+    incidentIds = _.pluck(@incidentIds.find().fetch(), 'id')
+    @subscribe "eventArticles", userEventId, incidentIds
+    if incidentIds
+      @subscribe 'eventIncidents', incidentIds, =>
```

Comment:
Putting these subscriptions in an autorun adds a round trip and slows down the load times. They could both take only the userEventId as a parameter and use that to figure out which incidents to return.